src/mpl/include/mpl_gpu.h

```diff
@@ -0,0 +1,36 @@
+/* -*- Mode: C; c-basic-offset:4 ; indent-tabs-mode:nil ; -*- */
+/* vim: set ft=c.mpich : */
+/*
+ *  (C) 2020 by Argonne National Laboratory.
+ *      See COPYRIGHT in top-level directory.
+ */
+
+#ifndef MPL_GPU_H_INCLUDED
+#define MPL_GPU_H_INCLUDED
+
+#include "mplconfig.h"
+
+#ifdef MPL_HAVE_GPU
+
+#define MPL_GPU_SUCCESS 0
+#define MPL_GPU_FAILURE 1
+
+typedef enum {
+    MPL_GPU_POINTER_HOST = 0,
+    MPL_GPU_POINTER_REGISTERED,
+    MPL_GPU_POINTER_DEV,
+    MPL_GPU_POINTER_MANAGED
+} MPL_pointer_type_t;
+
+int MPL_gpu_query_pointer_type(const void *ptr, MPL_pointer_type_t * attr);
+
+int MPL_gpu_ipc_get_mem_handle(MPL_gpu_ipc_mem_handle_t * h_mem, void *ptr);
+int MPL_gpu_ipc_open_mem_handle(void *ptr, MPL_gpu_ipc_mem_handle_t * h_mem);
+int MPL_gpu_ipc_close_mem_handle(void *ptr);
+
+int MPL_gpu_ipc_get_event_handle(MPL_gpu_ipc_event_handle_t * h_event, MPL_gpu_event_t * event);
+int MPL_gpu_ipc_open_event_handle(MPL_gpu_event_t * event, MPL_gpu_ipc_event_handle_t * h_event);
```

Comment:
What do you need events for at this point (at least till reductions are implemented)?

Reply:
I know these should be remove from the MPL wrapper PR. I have left a comment there too.